### DISCUSSION TITLE
GitHub Actions show pipeline failed with red cross while all jobs succeeded

### DISCUSSION THREAD

**Discussion by kuvaldini** [community/community#26760] [Actions]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/b/9ba4a2a186bd872f5724625ecc82741cf7f1db9c.png" data-download-href="/uploads/short-url/mcSJIsVjdloSGAv4prCxqZgPKFK.png?dl=1" title="изображение" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47349143/181099788-d73e9ee5-74a2-4eea-b4c5-8b732999b014.png" alt="изображение" data-base62-sha1="mcSJIsVjdloSGAv4prCxqZgPKFK" width="690" height="412" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/b/9ba4a2a186bd872f5724625ecc82741cf7f1db9c_2_690x412.png, https://aws1.discourse-cdn.com/github/optimized/3X/9/b/9ba4a2a186bd872f5724625ecc82741cf7f1db9c_2_1035x618.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/9/b/9ba4a2a186bd872f5724625ecc82741cf7f1db9c_2_1380x824.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/b/9ba4a2a186bd872f5724625ecc82741cf7f1db9c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">изображение</span><span class="informations">2974×1780 509 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<aside class="onebox githubactions" data-onebox-src="https://github.com/hyperledger/iroha/actions/runs/1007392564">
  <header class="source">

      <a href="https://github.com/hyperledger/iroha/actions/runs/1007392564" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">

      <svg width="24" height="24" class="github-icon github-icon-failure" viewbox="0 0 24 24" aria-hidden="true"><path d="M1 12C1 5.925 5.925 1 12 1s11 4.925 11 11-4.925 11-11 11S1 18.075 1 12zm8.036-4.024a.75.75 0 00-1.06 1.06L10.939 12l-2.963 2.963a.75.75 0 101.06 1.06L12 13.06l2.963 2.964a.75.75 0 001.061-1.06L13.061 12l2.963-2.964a.75.75 0 10-1.06-1.06L12 10.939 9.036 7.976z"></path></svg>

  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/hyperledger/iroha/actions/runs/1007392564" target="_blank" rel="noopener nofollow ugc">[CI][GHActions] Introduce GitHub Actions Iroha1 (#1028)
</a>
    </h4>

      <div class="github-info">
        Iroha1 <span class="github-run-number">#284</span>
      </div>
  </div>
</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Looks like jobs with empty matrices pending forever and workflow run is marked failed.</p>

**Comment by kuvaldini** [comment]
<p>here i found same problem <a href="https://github.community/t/empty-matrix-fails-workflow/128736" class="inline-onebox">Empty matrix fails workflow</a></p>

**Comment by kuvaldini** [comment, marked as answer]
<p>This is mine solution:</p>
<pre><code class="lang-auto">..........
    strategy:
      fail-fast: false
      matrix: ${{ fromJSON( needs.generate_matrixes.outputs.matrix_macos ) }}
    if: ${{ fromJSON( needs.generate_matrixes.outputs.matrix_macos ).include[0] }}
.........
</code></pre>

**Comment by wyarde** [comment]
<p>I am not sure what the <code>.include[0]</code> part is, but regardless I think you don’t need that; this should do:</p>
<pre><code class="lang-auto">..........
    strategy:
      fail-fast: false
      matrix: ${{ fromJSON( needs.generate_matrixes.outputs.matrix_macos ) }}
    if: ${{ needs.generate_matrixes.outputs.matrix_macos != '[]' }}
.........
</code></pre>
